pkg/resourceinterpreter/defaultinterpreter/healthy.go

```diff
@@ -0,0 +1,41 @@
+package defaultinterpreter
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/karmada-io/karmada/pkg/util"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+)
+
+type healthInterpreter func(object *unstructured.Unstructured) (bool, error)
+
+func getAllDefaultHealthInterpreter() map[schema.GroupVersionKind]healthInterpreter {
+	s := make(map[schema.GroupVersionKind]healthInterpreter)
+	s[appsv1.SchemeGroupVersion.WithKind(util.DeploymentKind)] = interpretDeploymentHealth
+	s[appsv1.SchemeGroupVersion.WithKind(util.StatefulSetKind)] = interpretStatefulSetHealth
+	return s
+}
+
+func interpretDeploymentHealth(object *unstructured.Unstructured) (bool, error) {
+	deploy := &appsv1.Deployment{}
+	err := helper.ConvertToTypedObject(object, deploy)
+	if err != nil {
+		return false, err
+	}
+
+	healthy := deploy.Status.AvailableReplicas == *deploy.Spec.Replicas
```

Comment:
I'd prefer the rule defined at [ArgoCD](https://argo-cd.readthedocs.io/en/stable/operator-manual/health/#deployment-replicaset-statefulset-daemonset):
```
Observed generation is equal to desired generation.
Number of updated replicas equals the number of desired replicas.
```

Reply:
Please add a unit test for this.